com.unity.render-pipelines.universal/Shaders/PostProcessing/UpscaleSetup.shader

```diff
@@ -0,0 +1,50 @@
+Shader "Hidden/Universal Render Pipeline/Upscale Setup"
+{
+    HLSLINCLUDE
+        #pragma multi_compile_local_fragment _ _FXAA
+        #pragma multi_compile _ _USE_DRAW_PROCEDURAL
+
+        #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
+        #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Filtering.hlsl"
+        #include "Packages/com.unity.render-pipelines.universal/ShaderLibrary/Core.hlsl"
+        #include "Packages/com.unity.render-pipelines.universal/Shaders/PostProcessing/Common.hlsl"
+
+        TEXTURE2D_X(_SourceTex);
+        float4 _SourceSize;
+
+        half4 Frag(Varyings input) : SV_Target
+        {
+            UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(input);
+
+            float2 uv = UnityStereoTransformScreenSpaceTex(input.uv);
+            float2 positionNDC = uv;
+            int2   positionSS = uv * _SourceSize.xy;
+
+            half3 color = SAMPLE_TEXTURE2D_X(_SourceTex, sampler_PointClamp, uv).xyz;
+
+#if _FXAA
+            color = ApplyFXAA(color, positionNDC, positionSS, _SourceSize, _SourceTex);
+#endif
+
+            return half4(color, 1.0);
```

Comment:
Is this Setup the "slow" path you mentioned in the design doc. The doc proposes the colorspace to be done inline with FXAA on the setup pass (just before FSR) but I don't see it here. I guess this will be changed in the FSR PR? (I haven't looked at it yet).

Reply:
Yes, this setup shader is the slow path I mentioned in the FSR doc. The color space conversion is indeed added into this shader in the FSR PR. I actually ended up removing the fast path earlier though based on some feedback from the URP team. When I did the optimization, I didn't realize that custom post processing passes are extremely common and the optimization can't work in the case where they're present. Based on that information, I figured it wasn't worth the additional code complexity cost. I could add it back in if we still want it? It's in this commit here: https://github.com/Unity-Technologies/Graphics/commit/26c523d90b0344af5e0edf1294fe7549d6789d56

Reply:
Ok. Can you point me to discussion, I guess I need some context to understand the pros / const of that. Othewise all good.

Reply:
I don't know if there's a written discussion on this one, it came out of a meeting we had earlier in November. I believe @Verasl was the one to point this out. You can see what the complexity cost in the commit. It's not particularly large, but it does cause logical coupling between uber post and final post, though I imagine the benefit of skipping a full screen pass could be pretty big on mobile so it might be worth having anyways even if it's usually skipped in most projects?